ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -429,24 +429,70 @@ func (r *RayJobReconciler) getOrCreateRayClusterInstance(ctx context.Context, ra
 	rayClusterInstance := &rayv1alpha1.RayCluster{}
 	err := r.Get(ctx, rayClusterNamespacedName, rayClusterInstance)
 	if err == nil {
-		r.Log.Info("RayJob associated rayCluster found", "rayjob", rayJobInstance.Name, "raycluster", rayClusterNamespacedName)
-		// just return the rayClusterInstance if cluster spec is nil
+		r.Log.Info("Found associated RayCluster for RayJob", "rayjob", rayJobInstance.Name, "raycluster", rayClusterNamespacedName)
+
+		// Case1: The job is submitted to an existing ray cluster, simply return the rayClusterInstance.
+		// We do not use rayJobInstance.Spec.RayClusterSpec == nil to check if the cluster selector mode is activated.
+		// This is because a user might set both RayClusterSpec and ClusterSelector. with rayJobInstance.Spec.RayClusterSpec == nil,
+		// though the RayJob controller will still use ClusterSelector, but it's now able to update the replica.
+		// this could result in a conflict as both the RayJob controller and the autoscaler in the existing RayCluster might try to update replicas simultaneously.
+		// if len(rayJobInstance.Spec.ClusterSelector) != 0 {
```

Comment:
```suggestion

```

remove this?

Reply:
Thank you for pointing out! I used for testing and forgot to remove it.